### PR TITLE
SC-200 Correct split()

### PR DESCRIPTION
**Pull request by xrtm000**

### CODE REVIEW

File: lang/jvm/src/test/scala/com/wavesplatform/lang/v1/evaluator/SplitFunctionTest.scala [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+         |
+      """.stripMargin
+
+    eval(script) shouldBe Right(CONST_BOOLEAN(true))
+  }
+
+  property("split separator around separator") {
+    val script =
+      s"""
+         |   let strs = split(",", ",")
```
Comment (msmolyakov) [inline lang/jvm/src/test/scala/com/wavesplatform/lang/v1/evaluator/SplitFunctionTest.scala]: add cases with ",,"

File: lang/jvm/src/test/scala/com/wavesplatform/lang/v1/evaluator/SplitFunctionTest.scala [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+         |                 strs1[3] == ""
+         |
+         |   let strs2 = split(";Q;Qsome;Q", ";Q")
+         |   let result2 = strs2.size() == 4  &&
+         |                 strs2[0] == ""     &&
+         |                 strs2[1] == ""     &&
+         |                 strs2[2] == "some" &&
+         |                 strs2[3] == ""
+         |
+         |   let strs3 = split("QQ;someQ;Q;;", "Q;")
```
Comment (msmolyakov) [inline lang/jvm/src/test/scala/com/wavesplatform/lang/v1/evaluator/SplitFunctionTest.scala]: change to `"q;Q;someQ;Q;;"`, expect `str3[0] == "q;"`

File: lang/jvm/src/test/scala/com/wavesplatform/lang/v1/evaluator/SplitFunctionTest.scala [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+         |
+      """.stripMargin
+
+    eval(script) shouldBe Right(CONST_BOOLEAN(true))
+  }
+
+  property("split empty string") {
+    val script =
+      s"""
+         |   let strs1 = split("", ",")
```
[the remark this answers is not in the document]
Comment (xrtm000) [reply lang/jvm/src/test/scala/com/wavesplatform/lang/v1/evaluator/SplitFunctionTest.scala]: +

File: lang/jvm/src/test/scala/com/wavesplatform/lang/v1/evaluator/SplitFunctionTest.scala (hunk 1 above)
Comment (xrtm000) [reply lang/jvm/src/test/scala/com/wavesplatform/lang/v1/evaluator/SplitFunctionTest.scala]: +

File: lang/jvm/src/test/scala/com/wavesplatform/lang/v1/evaluator/SplitFunctionTest.scala (hunk 2 above)
Comment (xrtm000) [reply lang/jvm/src/test/scala/com/wavesplatform/lang/v1/evaluator/SplitFunctionTest.scala]: +

**Pull request merged**